crates/uv-resolver/src/lock/mod.rs

```diff
@@ -613,6 +613,16 @@ impl Lock {
                 }
             }
 
+            // Validate any requested dependency groups.
+            for group in dev.groups().iter() {
+                if !root.metadata.requires_dev.contains_key(group) {
```

Comment:
I don't love using `metadata` here (rather than `root.dev_dependencies`) because this metadata is mostly just intended for lockfile validation / caching (we invalidate the lockfile when the metadata changes). But if we use `root.dev_dependencies`, we will _also_ fail if the user does `--group foo` when `foo` is empty. It seems wrong to fail when requesting a valid but empty group, so...